firstvoices/backend/models/characters.py

```diff
@@ -277,47 +256,77 @@ def preprocess_transducer(self):
             self.logger.warning("Empty confusable map for site %s", self.site)
             return None
 
-    @property
-    def presort_transducer(self):
+    def presort_transducer(self, context):
```

Comment:
Instead of passing a dictionary here, let's use named parameters for each specific thing and give them defaults of None. So something like 

def presort_transducer(self, base_characters=None, character_variants=None, etc)

Reply:
Same for the rest of the functions as well